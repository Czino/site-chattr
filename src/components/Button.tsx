import { ButtonHTMLAttributes } from 'react'

export const Button = (props: ButtonHTMLAttributes<HTMLButtonElement>) => (
    <button
        {...props}
        className={[
            'bg-gradient-purple text-white p-4 text-xl uppercase font-bold rounded hover:bg-gradient-purple-dark',
            props.className,
        ].join(' ')}
    ></button>
)
