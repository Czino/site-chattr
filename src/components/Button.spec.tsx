import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Button } from './Button'

describe('Button', () => {
    const onClick = jest.fn()
    it('should render Button with styles', () => {
        const { asFragment } = render(
            <Button className="mt-4" onClick={onClick}>
                Test
            </Button>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
    it('should call onClick handler on click', () => {
        render(
            <Button className="mt-4" onClick={onClick}>
                Test
            </Button>,
        )
        userEvent.click(screen.getByText('Test'))
        expect(onClick).toHaveBeenCalled()
    })
})
