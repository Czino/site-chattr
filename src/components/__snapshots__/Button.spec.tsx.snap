// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button with styles 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <button
-     class=\\"bg-gradient-purple text-white p-4 text-xl uppercase font-bold rounded hover:bg-gradient-purple-dark mt-4\\"
+     class=\\"bg-gradient-purple text-white p-4 text-xl uppercase font-bold rounded hover:bg-gradient-purple-dark \\"
    >
      Test
    </button>
  </DocumentFragment>"
`;

exports[`Button should render correctly 1`] = `
<DocumentFragment>
  <button
    class="bg-gradient-purple text-white p-4 text-xl uppercase font-bold rounded hover:bg-gradient-purple-dark "
  >
    Test
  </button>
</DocumentFragment>
`;
